File: hnaj-be/app/Enums/AuthErrorCode.php
<?php

namespace App\Enums;

/**
 * Mã lỗi ổn định cho các luồng authentication, dùng trong error envelope.
 */
enum AuthErrorCode: string
{
    case InvalidCredentials = 'INVALID_CREDENTIALS';   // Sai username hoặc password
    case EmailNotVerified = 'EMAIL_NOT_VERIFIED';      // Chưa xác thực email nên chưa hoàn tất đăng ký
    case AccountNotActive = 'ACCOUNT_NOT_ACTIVE';      // Tài khoản bị suspended hoặc disabled
    case InvalidToken = 'INVALID_VERIFICATION_TOKEN';  // Token verify sai, đã dùng hoặc hết hạn
    case AlreadyVerified = 'EMAIL_ALREADY_VERIFIED';   // Email đã được xác thực trước đó
    case Unauthenticated = 'UNAUTHENTICATED';          // Thiếu hoặc sai bearer token
    case ForbiddenRole = 'FORBIDDEN_ROLE';             // Thiếu role bắt buộc cho endpoint
    case GoogleAuthFailed = 'GOOGLE_AUTH_FAILED';      // Không hoàn tất được OAuth với Google
}

File: hnaj-be-java/src/main/java/com/hnaj/api/exception/AuthErrorCode.java
package com.hnaj.api.exception;

/**
 * Mã lỗi ổn định cho luồng authentication — mapping 1-1 với
 * hnaj-be/app/Enums/AuthErrorCode.php.
 * Xem docs/migration/knowledge-base/01-api-contract.md §4.
 */
public enum AuthErrorCode {
    INVALID_CREDENTIALS,         // 401
    EMAIL_NOT_VERIFIED,          // 403
    ACCOUNT_NOT_ACTIVE,          // 403
    INVALID_VERIFICATION_TOKEN,  // 422
    EMAIL_ALREADY_VERIFIED,      // 409
    UNAUTHENTICATED,             // 401
    FORBIDDEN_ROLE,              // 403
    GOOGLE_AUTH_FAILED           // 422
}
